prow/plugins/plugins.go

```diff
@@ -191,6 +208,21 @@ func (c *Configuration) TriggerFor(org, repo string) *Trigger {
 	return nil
 }
 
+func (c *Configuration) setDefaults() {
```

Comment:
I see how this is making the change non-breaking, but do we want this pattern?

Reply:
We already do this for the main config file, see [here](https://github.com/kubernetes/test-infra/blob/b4fe4eda5931ddb3c43ebfa67932b4303d7e532f/prow/config/config.go#L202-L246).

Reply:
Can we try to localize this to the config structs? Not sure hundreds of LOC for defaulting all sorts of unrelated things would be nice in one place

Reply:
We would still need to have all the defaulting in a single place because otherwise it would be a madness. We should probably refactor the prow config to have a separate function for defaulting that includes all the code I linked above but in a separate PR.